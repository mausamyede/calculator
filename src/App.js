import './App.css';
import React, {Component} from "react";
import { v4 as uuidv4 } from 'uuid';

class App extends Component {
    constructor(props) {
        super(props);
        this.state = {
            expression: '',
            logs: ''
        }
        this.onEvaluate = this.onEvaluate.bind(this);
        this.onTextChange = this.onTextChange.bind(this);
    }

    componentDidMount() {
        let browserName = "Browser"+uuidv4();
        this.connection = new WebSocket("wss://calculator-app-backend.herokuapp.com/evaluate?name="+browserName);
        this.connection.onmessage = event => {
            this.setState({
                logs: event.data
            })
        }
    }

    onTextChange(e) {
        this.setState({
            expression: e.target.value
        })
    }

    onEvaluate() {
        this.connection.send(this.state.expression)
    }

    render() {
        return (
            <div className="App">
                <h1 className="title">Shared Calculator</h1>
                <div className="Input-Box">
                    <input type="text" value={this.state.expression} onChange={this.onTextChange}/>
                    <button onClick={this.onEvaluate}>Evaluate</button>
                </div>
                <div className="Log-Box">
                    <textarea name="logs" rows="10" cols="50" value={this.state.logs}/>
                </div>
            </div>
        )
    }
}

export default App;
