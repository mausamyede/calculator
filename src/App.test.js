import React from 'react';
import {configure, shallow} from 'enzyme';
import App from "./App";
import Adapter from 'enzyme-adapter-react-16';

configure({adapter: new Adapter()});

describe('App', () => {
    it('should render correctly', () => {
        const component = shallow(<App/>);

        expect(component).toMatchSnapshot();
    });
});
