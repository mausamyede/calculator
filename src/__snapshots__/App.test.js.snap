// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
<div
  className="App"
>
  <h1
    className="title"
  >
    Shared Calculator
  </h1>
  <div
    className="Input-Box"
  >
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
    <button
      onClick={[Function]}
    >
      Evaluate
    </button>
  </div>
  <div
    className="Log-Box"
  >
    <textarea
      cols="50"
      name="logs"
      rows="10"
      value=""
    />
  </div>
</div>
`;
